src/Microsoft.IdentityModel.Tokens/EventBasedLRUCache.cs

```diff
@@ -307,11 +333,14 @@ public bool SetValue(TKey key, TValue value, DateTime expirationTime)
             {
                 cacheItem.Value = value;
                 cacheItem.ExpirationTime = expirationTime;
-                AddActionToEventQueue(() =>
+                if (_maintainLRU)
```

Comment:
@dannybtsai I am not understanding how the map is compacted.
The method RemoveLRUs() checks the _doubleLinkedList.Count>0 on line 291, but i don't see where an item is added to _doubleLinkedList ?

Reply:
Danny and i reviewed this and it seems fine, i just needed some hand holding....

However, we decided, that is we are using the MRU part, just skip adding to the _doubleLinkList and modify the RemoveLRU to only consider the _doubleLinkList if MRU logic is in.

We also suggested to rename RemoveLRU to Compact.